tgui/packages/tgui/interfaces/DJPanel.js

```diff
@@ -0,0 +1,215 @@
+              onClick={() => act('play-player')}
+            />
+          </Box>
+          <Box>
+            <Button
+              color="yellow"
+              content="Toggle DJ Announcements"
+              onClick={() => act('toggle-announce')}
+            />
+            <Button
+              color="yellow"
+              content="Toggle DJ For Player"
+              onClick={() => act('toggle-player-dj')}
+            />
+          </Box>
+          <Box>
+            <Button
+              icon="stop"
+              color="red"
+              content="Stop Last Sound"
+              onClick={() => act('stop-sound')}
+            />
+            <Button
+              icon="broadcast-tower"
+              color="red"
+              content="Stop The Radio For Everyone"
+              onClick={() => act('stop-radio')}
+            />
+          </Box>
+        </Section>
+        <AnnounceActive />
+      </Window.Content>
+    </Window>
+  );
+};
+
+
+const AnnounceActive = (props, context) => {
+  const { data } = useBackend(context);
+  const { announceMode } = data;
+
+  if (announceMode) {
+    return (
+      <NoticeBox info>
+        Announce Mode Enabled
+      </NoticeBox>
+    );
+  }
+};
+
+const KnobZone = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { loadedSound, volume, frequency } = data;
+
+  return (
+    <Box>
+      <LabeledControls>
+        <LabeledControls.Item label="Volume">
+          <NumberInput
+            animated
+            value={volume}
+            minValue={0}
+            maxValue={100}
+            format={value => {
+              return toFixed(value * 2) + '%';
+            }}
+            onDrag={(e, value) => act('set-volume', {
+              volume: value,
+            })}
+          />
+        </LabeledControls.Item>
+        <LabeledControls.Item>
+          <Knob
+            minValue={0}
+            maxValue={100}
+            ranges={{
+              primary: [20, 80],
+              average: [10, 90],
+              bad: [0, 100],
+            }}
+            value={volume}
+            format={value => {
+              return toFixed(value * 2) + '%';
+            }}
+            onDrag={(e, value) => act('set-volume', {
+              volume: value,
+            })}
+          />
+          <Button
+            icon="sync-alt"
+            top="0.3em"
+            content="Reset"
+            onClick={() => act('set-volume', {
+              volume: "reset",
+            })}
+          />
+        </LabeledControls.Item>
+        <LabeledControls.Item label="Frequency">
+          <NumberInput
+            animated
+            value={frequency}
+            step={0.1}
+            minValue={-100}
+            maxValue={100}
+            format={value => {
+              return toFixed(value * 100) + '%';
+            }}
+            onDrag={(e, value) => act('set-freq', {
+              frequency: value,
+            })}
```

Comment:
As these are re-used for the `Knob` below, would prefer they're created outside of the `return` and then referenced both here and in the `Knob`'s props.